runtime/gateway.go

```diff
@@ -440,20 +440,20 @@ func (gateway *Gateway) setupTChannel(config *StaticConfig) error {
 	serviceName := config.MustGetString("tchannel.serviceName")
 	processName := config.MustGetString("tchannel.processName")
 
+	subScope := gateway.MetricScope.SubScope("tchannel")
```

Comment:
Can we write a test for http endpoint + tchannel client and write assertions on what metrics are emitted ?

We should also re-run the benchmarks to see what the performance implications are and make some flamegraphs to double check if there are any low hanging fruit.